src/GitVersion.Core/VersionCalculation/NonTrunkBasedVersionCalculatorBase.cs

```diff
@@ -0,0 +1,93 @@
+using GitVersion.Common;
+using GitVersion.Configuration;
+using GitVersion.Extensions;
+using GitVersion.Logging;
+
+namespace GitVersion.VersionCalculation;
+
+internal abstract class NonTrunkBasedVersionCalculatorBase
+{
+    protected readonly ILog log;
+    protected readonly IRepositoryStore repositoryStore;
+    private readonly Lazy<GitVersionContext> versionContext;
+
+    protected GitVersionContext Context => this.versionContext.Value;
+
+    public NonTrunkBasedVersionCalculatorBase(ILog log, IRepositoryStore repositoryStore, Lazy<GitVersionContext> versionContext)
+    {
+        this.log = log.NotNull();
+        this.repositoryStore = repositoryStore.NotNull();
+        this.versionContext = versionContext.NotNull();
+    }
+
+    protected bool ShouldTakeIncrementedVersion(NextVersion nextVersion)
+    {
+        var label = nextVersion.Configuration.GetBranchSpecificLabel(
+            Context.CurrentBranch.Name, nextVersion.BaseVersion.BranchNameOverride
+        );
+
+        ////
+        // TODO: We need to decide whether or not to calculate the upcoming semantic version or the previous one because of tagging. Actually this should be part of the branch configuration system.
```

Comment:
Agreed. Can you please add this as an issue we can follow up on?

Reply:
I think the issue is already existent: https://github.com/GitTools/GitVersion/issues/3453